### PR TITLE
Improve deletion of entities

### PR DESCRIPTION
**Pull request by BryanValverdeU**
Improve the Content Edit Feature to delete delimiters.
 If After the Content Edit was handled:
1. If the Entity still exists, add the delimiter removed.
2. If the entity was removed in the ContentChanged Event, remove the Delimiters.
3. If the event was not default prevented, select the delimiters and entity and let the browser handle the deletion.

Also, in EntityPlugin if the known entity removed is a inline entity, normalize delimiters

### CODE REVIEW

File: packages/roosterjs-editor-core/lib/corePlugins/EntityPlugin.ts [hunk 1]
```diff
@@ -236,6 +241,14 @@ export default class EntityPlugin implements PluginWithState<EntityPluginState>
                 if (element.shadowRoot) {
                     this.triggerEvent(element, EntityOperation.RemoveShadowRoot);
                 }
+
+                if (
+                    !shouldNormalizeDelimiters &&
+                    !element.isContentEditable &&
+                    getTagOfNode(element) === 'SPAN'
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-core/lib/corePlugins/EntityPlugin.ts:248]: Why check SPAN?

File: packages/roosterjs-editor-dom/lib/index.ts [hunk 2]
```diff
@@ -4,11 +4,7 @@ export { default as getFirstLastBlockElement } from './blockElements/getFirstLas
 export { default as ContentTraverser } from './contentTraverser/ContentTraverser';
 export { default as PositionContentSearcher } from './contentTraverser/PositionContentSearcher';
 
-export {
-    default as addDelimiters,
-    addDelimiterAfter,
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-dom/lib/index.ts:9]: This is a 
breaking change?

File: packages/roosterjs-editor-core/lib/corePlugins/EntityPlugin.ts (hunk 1 above)
Comment (BryanValverdeU) [reply packages/roosterjs-editor-core/lib/corePlugins/EntityPlugin.ts:248]: To only handle the normalization if the deleted entity is readonly and inline

File: packages/roosterjs-editor-dom/lib/index.ts (hunk 2 above)
Comment (BryanValverdeU) [reply packages/roosterjs-editor-dom/lib/index.ts:9]: Hmm, fixing that

File: packages/roosterjs-editor-core/lib/corePlugins/EntityPlugin.ts (hunk 1 above)
Comment (BryanValverdeU) [reply packages/roosterjs-editor-core/lib/corePlugins/EntityPlugin.ts:248]: Agree, that might be better.

**Review by JiuqingSong — APPROVED** [review]

**Pull request merged**